### ISSUE TITLE
Unhanded Error: spawn while updating

### ISSUE THREAD

**Issue by blimmo** [buttercup/buttercup-desktop#690]
Using the auto-updater I get this:

![image](https://user-images.githubusercontent.com/26982890/48551722-40086880-e8ce-11e8-8807-11884df1fd28.png)

Started in 1.10.3 and I hoped it would sort itself out by the next update but it's still failing on 1.11.0

**Comment by perry-mitchell** [comment]
@blimmo Can you add/edit/remove files within `C:\Users\Peter\AppData\Local\Temp\` ? Would be good to check if you can to see what the problem is. ACCES is obviously a permissions issue with your machine.

**Comment by perry-mitchell** [comment]
Is it possible that Buttercup was installed with raised permission levels, like from an admin user or super user privileges?

Would you mind uninstalling and reinstalling from a freshly-loaded version of say [1.10.3](https://github.com/buttercup/buttercup-desktop/releases/tag/v1.10.3) so we can test the update functionality?

**Comment by blimmo** [comment]
Reinstalled from 1.10.3 and no change.

I then tried 1.10.1 (last version autoupdates worked) and this worked! I'm now on 1.11.0.

It seems that it's just updating *from* 1.10.2-3 (and possibly 1.11.0 but I can't test that yet) that's the problem.

I'm happy to keep testing things if you need me to :)

**Comment by sallar** [comment]
@blimmo @perry-mitchell I accidentally commented about this on the wrong thread 😂 https://github.com/electron-userland/electron-builder/issues/1580#issuecomment-439060293 but that question is for @blimmo actually. Does this happen when you click on Download or does it happen when the app tries to check for updates?

You can see that this is the same error that is mentioned in that issue.

**Comment by blimmo** [comment]
It's when I click download. I got an elevation prompt when it worked but the failing times seem to have missed that out somehow.

**Comment by jenstornell** [comment]
I also get this all the time since a while back in time. So this time I'll speak out loud here.

- It's the exact same error as the screenshot.
- It's when pressing download as stated.
- I tried to install it manually from the exe file. That worked, but new versions are released quite often so it's not a long term solution.
- I'm on Windows 10.

**Comment by perry-mitchell** [comment]
Also reproducible for me now:

![image](https://user-images.githubusercontent.com/3869469/49104302-4ca19f00-f286-11e8-9e01-b3640eb258fb.png)

**Comment by n8v8R** [comment]
Same happened on my W10 node and had to download/run the installer manually. This might be a W10 security measure. Defender did not recognise the manually downloaded installer and produces a warning.

**Comment by perry-mitchell** [comment]
As mentioned earlier it can be run from the temp folder - it causes a permission dialog to show and then it works normally. Running it from the command line fails as does the update process in the app. I think you’re right about it being a security precaution.

Definitely time to have the windows stuff signed properly I think.

**Comment by jenstornell** [comment]
I agree! I thought it was a bit weird to get a warning from Windows that this app may not be safe, because it's an app for just that, security. ☺

**Comment by perry-mitchell** [comment]
I'd say that this is then blocked by #704

**Comment by n8v8R** [comment]
Just for completeness the Defender SmartScreen warning

![defender smart screen warning](https://user-images.githubusercontent.com/13611490/49182063-1ed55c80-f351-11e8-8f48-5752100e9739.png)

**Comment by MiloradFilipovic** [comment]
I managed to update it once I run buttercup as administrator.

**Comment by d2s** [comment]
Tried to upgrade Buttercup Desktop (with the built-in update check)
from the version 1.11.x to version 1.13.0, and got the following error message.

```
Unhandled Error
Error: spawn C:/Users/UserName/AppData/Local/Temp/up-0t8Xtm/buttercup-desktop-setup-1.13.0.exe EACCES
```

Looked closer to the installer file in the directory
`C:/Users/UserName/AppData/Local/Temp/up-0t8Xtm/`
and found out that it seems to work properly when started directly.

Understandably that this is related to the Windows code signing that you are working on to fix.

**Comment by perry-mitchell** [comment]
Really appreciate the help and patience here. @sallar and I are looking at investing in a codesigning cert shortly so we can clear up some of these issues. We'll hopefully take a look at it tomorrow and over the next few days.

**Comment by n8v8R** [comment]
Since the code now being signed with v1.14.0 I guess will have to wait until the next update rolls out and see whether it sorts this issue.

Defender's SmartScreen seems not impressed though. Maybe the certificate needs to get propagated into the MS database. 

![defender smart screen](https://user-images.githubusercontent.com/13611490/50606644-62763580-0ebe-11e9-8b06-47d890ef901b.PNG)

**Comment by blimmo** [comment]
Tried to update to 1.14.0 and the error still occurred. I can try updating manually this time and then hopefully future updates will work?

**Comment by jy95** [comment]
Same for 1.15.0 : why not allow user to set up a custom download directory ?

**Comment by n8v8R** [comment]
confirmed

![1 1 5](https://user-images.githubusercontent.com/13611490/52751889-40a9bb80-2ff1-11e9-95d0-85bb12e950db.png)

**Comment by Blackfaded** [comment]
Having the same issue with 1.15.0 -> 1.16.2
Run application as administrator did work!

**Comment by Jib-san** [comment]
Whenever an update is released, I'm running the application as administrator , just for the update.

Like others app, can't it just asking administrator's rights for upgrade time ?

**Comment by perry-mitchell** [comment]
We’re using the update system that came with the platform, electron. While it could be a flag we’re missing, I attribute the issue to the update system not being a great fit for us. If we can’t easily find a fix for it we’ll probably turf it and find a new one..
